### PR TITLE
Reserve mission docs

### PR DESCRIPTION
**Pull request by ehariton**
### Summary

Companion to Reserve Mission PR, contains docs and graphical examples of reserve mission usage and output.

### Related Issues

- Resolves #

### Backwards incompatibilities

None

### New Dependencies

None

### CODE REVIEW

**Review by johnjasa — APPROVED** [review]
I pushed some simple changes directly to the PR! Great stuff!

File: aviary/docs/examples/reserve_missions.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Reserve Mission Examples
+
+## Fixed Range Reserve
+
+The simplest example of a reserve mission implementation can be seen in `run_reserve_mission_fixedrange.py`.
+In this example we add a fixed range cruise to the `run_basic_aviary_example.py`.
+This is performed by copying the cruise `phase_info` object from the basic example and adding `"reserve": True,` and `"target_distance": (200, 'km'),`.
+
+The results of this study are shown in the `traj_results_report.html`.
+The first things we review is the altitude plot.
```
Comment (crecine) [inline aviary/docs/examples/reserve_missions.md:10]:
```suggestion
The first thing we review is the altitude plot.
```

File: aviary/docs/examples/reserve_missions.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+We make a copy of the cruise phase from `example_phase_info.py` and then modify it by adding `"reserve": True,` and `"target_duration": (30, 'min'),`.
+We have also removed other references to time in the reserve phase definition.
+We do not list `initial_guesses:times`, `user_options:duration_bounds`, or `user_options:fixed_duration`.
+All of these are set automatically when we set `target_duration`.
+
+## Multiphase Reserve Mission
+
+The last example reserve mission includes five phases: climb, cruise for distance, cruise for time, cruise for distance, descend.
+This example is contained in `run_reserve_mission_multiphase.py`.
+This demonstrates how to create an arbitrarily ordered set of phases for reserve in addition to climb and descent phases.
```
Comment (crecine) [inline aviary/docs/examples/reserve_missions.md:35]:
```suggestion
This demonstrates how to create an arbitrarily ordered set of phases for reserve, including climb and descent phases.
```

File: aviary/docs/user_guide/reserve_missions.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+## Overview
+
+Reserve missions are enabled for the following mission types:
+
+* height_energy    (completed)
+* 2ODF (collocation) (in-progress)
+* 2DOF (shooting)    (future work)
+
+Reserve missions can be used to create fixed-time or a fixed-distance phases that are appended to the last phase of your nominal mission.
+To do this, set the `"reserve": True,` flag on within the `user_options` for your reserve phases within `phase_info`.
```
Comment (crecine) [inline aviary/docs/user_guide/reserve_missions.md:12]:
```suggestion
To do this, set the `"reserve": True,` flag within the `user_options` for your reserve phases within `phase_info`.
```

File: aviary/docs/user_guide/reserve_missions.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+## Examples
+
+Examples of single-phase and multi-phase reserve missions are presented in [Reserve Mission Examples](../examples/reserve_missions.md).
+
+## Theory
+
+When adding a reserve phase, `check_and_preprocess_inputs()` divides all the phases into two dictionaries: `regular_phases` which contain your nominal phases and `reserve_phases` which contains any phases with the `reserve` flag set to `True`.
+
+Only the final mission mass and range from `regular_phases` is automatically connected to the first point of the `reserve_phases`.
```
Comment (crecine) [inline aviary/docs/user_guide/reserve_missions.md:34]:
```suggestion
Only the final mission mass and range from `regular_phases` are automatically connected to the first point of the `reserve_phases`.
```

**Review by crecine — APPROVED** [review]

**Pull request merged**